### PR TITLE
fix: allow namespaces to be empty with `zarf tools wait-for`

### LINKED ISSUE #2054: Zarf 0.30.0 wait-for not working in some scenarios
### Environment
Device and OS: Macos, MBP
App version: 0.30.0
Kubernetes distro being used: k3d

### Steps to reproduce

Option 1:

- Build/deploy the below package with zarf 0.29.2 and zarf 0.30.0, against a default k3d cluster:

```yaml
kind: ZarfPackageConfig
metadata:
  name: test
components:
  - name: bar
    actions:
      onDeploy:
        after:
          - wait:
              cluster:
                kind: APIService
                name: v1beta1.metrics.k8s.io
```

Option 2:

- Run the following commands against a default k3d cluster:

```console
❯ zarf-30 tools wait-for apiservice v1beta1.metrics.k8s.io -n default
  ✔  Waiting for apiservice/v1beta1.metrics.k8s.io in namespace default to exist.                                                                                                                               
❯ zarf-30 tools wait-for apiservice v1beta1.metrics.k8s.io -n ""
  ⠦  Waiting for apiservice/v1beta1.metrics.k8s.io to exist. (3s)^C                                                                                                                                             
❯ zarf-30 tools wait-for apiservice v1beta1.metrics.k8s.io
  ⠦  Waiting for apiservice/v1beta1.metrics.k8s.io to exist. (3s)^C                                                                                                                                             
```

(the first one passes, the others fail)

### Expected result

The wait-for behavior should be the same across these versions.

### Actual Result

Wait-for seems to fail in some scenarios, possibly due to namespace handling behavior with the kubectl library update?

### Visual Proof (screenshots, videos, text, etc)

Included logs above for CLI behavior.

### Severity/Priority

Breaks the BB extension due to the metrics server wait included there.

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This fixes an issue with waiting for a namespace when a namespace isn't specified.

## Related Issue

Fixes #2054

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/test/e2e/31_component_action_cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+func TestComponentActionEdgeCases(t *testing.T) {
+	t.Log("E2E: Component action edge cases")
+	e2e.SetupWithCluster(t)
+
+	sourcePath := filepath.Join("src", "test", "packages", "31-component-actions-edgecases")
+	packagePath := fmt.Sprintf("zarf-package-component-actions-edgecases-%s.tar.zst", e2e.Arch)
+
+	stdOut, stdErr, err := e2e.Zarf("package", "create", sourcePath, "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+
```
Comment (lucasrod16) [inline src/test/e2e/31_component_action_cluster_test.go:41]: Although this test package is minimal, it would still probably be good to remove the package to help with our disk pressure issues in CI since they can add up over time if left on disk

```suggestion
defer e2e.CleanFiles(packagePath)

```

**Review by lucasrod16 — CHANGES_REQUESTED** [review]

File: src/test/e2e/31_component_action_cluster_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+func TestComponentActionEdgeCases(t *testing.T) {
+	t.Log("E2E: Component action edge cases")
+	e2e.SetupWithCluster(t)
+
+	sourcePath := filepath.Join("src", "test", "packages", "31-component-actions-edgecases")
+	packagePath := fmt.Sprintf("zarf-package-component-actions-edgecases-%s.tar.zst", e2e.Arch)
+
+	stdOut, stdErr, err := e2e.Zarf("package", "create", sourcePath, "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+defer e2e.CleanFiles(packagePath)
```
Comment (lucasrod16) [inline src/test/e2e/31_component_action_cluster_test.go:41]:
```suggestion
    defer e2e.CleanFiles(packagePath)
```

**Review by lucasrod16 — COMMENTED** [review]
I didn't use correct tabbing on the previous suggestion

otherwise this lgtm

**Review by lucasrod16 — APPROVED** [review]

**Review by Noxsios — APPROVED** [review]
